src/org/jitsi/impl/neomedia/transform/PaddingTermination.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright @ 2015 Atlassian Pty Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.impl.neomedia.transform;
+
+import org.jitsi.impl.neomedia.*;
+import org.jitsi.service.neomedia.*;
+
+import java.util.*;
+
+/**
+ * De-duplicates RTP packets from incoming RTP streams.
+ *
+ * @author George Politis
+ */
+public class PaddingTermination
+    extends SinglePacketTransformerAdapter
+    implements TransformEngine
+{
+    /**
+     * Ctor.
+     */
+    public PaddingTermination()
+    {
+        super(RTPPacketPredicate.INSTANCE);
+    }
+
+    /**
+     * The {@code ReplayContext} for every SSRC that this instance has seen.
+     */
+    private final Map<Integer, ReplayContext> replayContexts = new TreeMap<>();
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PacketTransformer getRTPTransformer()
+    {
+        return this;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public PacketTransformer getRTCPTransformer()
+    {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public RawPacket reverseTransform(RawPacket pkt)
+    {
+        Integer mediaSSRC = pkt.getSSRC();
+
+        // TODO maybe drop padding from the main RTP stream?
+        ReplayContext replayContext = replayContexts.get(mediaSSRC);
```

Comment:
Yeah.. I'm sorry for the confusion, the `PaddingTermination` class only performs deduplication. Padding packets in the RTX stream are killed in the `RtxTransformer` and padding packet in the media stream are not killed at all (it's not easy to kill them, so one of the reasons why we wanted RTX to begin with). tl;dr; I need to find a more appropriate name for PaddingTermination... Originally I had named this `DedupTransformer`..